kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ConflictChecker.java

```diff
@@ -191,6 +212,71 @@ private void handleMetadata(ColumnVector metadataVector) {
     }
   }
 
+  /**
+   * Checks whether each of the current transaction's {@link DomainMetadata} conflicts with the
+   * winning transaction at any domain.
+   *
+   * <ol>
+   *   1. Accept the current transaction if its set of metadata domains does not overlap with the
+   *   winning transaction's set of metadata domains.
+   *   <p>2. Otherwise, fail the current transaction unless each conflicting domain is associated
+   *   with a domain-specific way of resolving the conflict.
+   * </ol>
+   *
+   * @param actionBatch action batch from the winning transactions
+   */
+  private void handleDomainMetadata(ColumnarBatch actionBatch) {
+    // Extract the domain metadata map from the winning transaction.
+    Map<String, DomainMetadata> winningDomainMetadataMap =
+        DomainMetadataUtils.extractDomainMetadataMap(
+            inMemoryIterable(actionBatch.getRows()), CONFLICT_RESOLUTION_SCHEMA);
```

Comment:
Why do you need an iterarable? (Something that is able to yield iterators). Why not just an iterator?

Reply:
Like mentioned above, I originally tried to reuse `extractDomainMetadataMap` in several places, so the API was designed that way (though, in retrospect, using an iterator would have been better).

Now that `extractDomainMetadataMap` is only used within `ConflictChecker`, I’ve updated this part and `extractDomainMetadataMap` will just require an `ColumnVector`.